client/http/witness_client.go

```diff
@@ -93,6 +93,9 @@ func (w Witness) Update(ctx context.Context, logID string, cp []byte, proof [][]
 	if err != nil {
 		return nil, fmt.Errorf("failed to do http request: %v", err)
 	}
+	if resp.Request.Method != "PUT" {
```

Comment:
```suggestion
	if resp.Request.Method != http.MethodPut {
```

Reply:
I was going to do this in a follow-up PR, but I've added it as a separate commit to this one. PTAL?